Software CLI Reference/ Command Updates (#709)
astronomer/docs: software/cli-reference.md

### Flags

| Flag                        | Value Type | Usage                                                                                                                    |
| --------------------------- | ---------- | ------------------------------------------------------------------------------------------------------------------------ |
| `--deployment-id`           | String     | The ID of the Deployment for which you want to upgrade Airflow. To find your Deployment ID, run `astro deployment list`. |

> | `--desired-airflow-version` | String     | The Airflow version you're upgrading to (e.g. `1.10.14`).                                                                |

Reviewer: ```suggestion
| `--desired-airflow-version` | String     | The Airflow version you're upgrading to (e.g. `2.2.0`).                                                                |
```

### Flags

| Flag                         | Value Type | Usage                                                                                                                           |
| ---------------------------- | ---------- | ------------------------------------------------------------------------------------------------------------------------------- |
| `--category`                 | String     | The category for the new service account as displayed in the Software UI. This is optional, and the default value is `Not set`. |
| `--deployment-id` (Required) | String     | The Deployment you're creating a service account for.                                                                           |
| `--label` (Required)         | String     | The name or label for the new service account.                                                                                  |

> | `--role`                     | String     | The User Role for the new service account. Can be `viewer`, `editor`, or `admin`. The default value is `viewer`.                |

Reviewer: Did we delete `--user-id`?

Author: yup, per https://github.com/astronomer/cloud-cli/blob/cli-convergence/CONVERGENCE_CHANGES.md#deployment-service-account-commands

| ----------------------- | ---------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `--cloud-role`          | String     | The ARN for the IAM role.                                                                                                                                                                                     |
| `--dag-deployment-type` | String     | The DAG deploy method for the Deployment. Can be either `image` or `volume`. The default value is `image`.                                                                                                    |
| `--nfs-location`        | String     | The location for an NFS volume mount, specified as: `<IP>:/<path>`. Must be specified when `--dag-deployment-type=volume`. Input is automatically prepended with `nfs:/` - do not include this in your input. |

> | `label`                 | String     | The label for the Deployment.                                                                                                                                                                                 |

Reviewer: ```suggestion
| `label`                 | String     | The label or name for the Deployment.                                                                                                                                                                                 |
```

## astro dev init

> Initializes a new Airflow project in your working directory. The set of files generated by this command are required to run Airflow locally and can be pushed to a Deployment on Astronomer.

Reviewer: ```suggestion
Initializes a new Astro project in your working directory. The set of files generated by this command are required to run Airflow locally and can be pushed to a Deployment on Astronomer.
```

:::info

> If you want to deploy your project to a Software Deployment, you must either specify the `--use-astronomer-certified` flag when running `astro dev init` or update the image in your project's `Dockerfile` to a supported Astronomer Certified image.

Reviewer: ```suggestion
To deploy your project to a Software Deployment, you must either specify the `--use-astronomer-certified` flag when running `astro dev init` or update the image in your project's `Dockerfile` to a supported Astronomer Certified image. For more information, see [Deploy DAGs via Astro CLI](deploy-cli.md).
```

### Flags

| Flag                         | Value Type                                                                                                                                                      | Usage                                                                                              |
| ---------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------------- |

> | `--airflow-version`          | String                                                                                                                                                          | The version of Airflow you want to use. The default value is the latest Airflow version available. |

Reviewer: ```suggestion
| `--airflow-version`          | String                                                                                                                                                          | The version of Airflow you want to use. The default value is the latest Airflow version available for Astronomer Certified. |
```

> | `--use-astronomer-certified` | `` | Create the new project with the latest version of Astronomer Certified. By default, your project is created with the latest version of Astronomer Runtime. |

Reviewer: ```suggestion
| `--use-astronomer-certified` | `` | Create the new project with the latest version of Astronomer Certified. By default, your project is created with the latest version of Astronomer Certified. |
```

- [Running the Airflow Upgrade Check Package](https://airflow.apache.org/docs/apache-airflow/stable/upgrade-check.html#upgrade-check)

## astro login/ logout

> Authenticates you to Astronomer.

Reviewer: ```suggestion
Authenticates you into and out of an installation on Astronomer Software.
```

### Usage

Run `astro login <base-domain>` or `astro logout <base-domain>` to log in or out of your Astronomer platform respectively. This is equivalent to using the login screen of the Software UI.

> If you have access to more than one Astronomer platform, each will have a unique `<base-domain>`. When switching between platforms, make sure to log out of one `<base domain>` before logging into another.

Reviewer: ```suggestion
If you have access to more than one Astronomer installation, each will have a unique `<base-domain>`. When switching between platforms, make sure to log out of one `<base domain>` before logging into another.
```

### Usage

> `astro workspace create --label=<new-workspace-name> [flags]`

Reviewer: Is this `--name` or `--label`? The options below list if as `--name` .Assume it's `--label` - let's make sure everything below is consistent

cc: @sunkickr

Author: It's label per https://github.com/astronomer/cloud-cli/blob/cli-convergence/CONVERGENCE_CHANGES.md#workspace-commands
